packages/language-service/src/lib/css-service.ts

```diff
@@ -128,59 +129,85 @@ export class CssService {
                     readDocRange(
                         document,
-                        Range.create(Position.create(diag.range.start.line, 0), diag.range.end)
+                        Range.create(Position.create(diagStart.line, 0), diagEnd)
                     ).startsWith('@custom-selector')
                 ) {
                     return false;
-                }
-                if (diag.code === 'css-rparentexpected' || diag.code === 'css-identifierexpected') {
+                } else if (
+                    diag.code === 'css-rparentexpected' ||
+                    diag.code === 'css-identifierexpected'
+                ) {
                     const line = readDocRange(
                         document,
-                        Range.create(Position.create(diag.range.start.line, 0), diag.range.end)
+                        Range.create(Position.create(diagStart.line, 0), diagEnd)
                     );
-                    const stateStart = findPseudoStateStart(line, diag.range.start.character);
+                    const stateStart = findPseudoStateStart(line, diagStart.character);
 
                     if (stateStart.index !== -1 && stateStart.openParens > 0) {
                         return false;
                     }
-                }
-                if (diag.code === 'unknownProperties') {
+                } else if (diag.code === 'unknownProperties') {
                     const prop = diag.message.match(/'(.*)'/)![1];
 
-                    const uri = URI.parse(document.uri);
-                    // on windows, uri.fsPath replaces separators with '\'
-                    // this breaks posix paths in-memory when running on windows
-                    // take raw posix path instead
-                    const filePath =
-                        uri.scheme === 'file' &&
-                        !uri.authority && // not UNC
-                        uri.path.charCodeAt(2) !== 58 && // the colon in "c:"
-                        path.isAbsolute(uri.path)
-                            ? uri.path
-                            : uri.fsPath;
-
-                    const src = this.fs.readFileSync(filePath, 'utf8');
-                    const meta = createMeta(src, filePath).meta;
-                    if (meta && Object.keys(meta.getAllSymbols()).some((ms) => ms === prop)) {
+                    if (meta.getStVar(prop)) {
                         return false;
                     }
+                } else if (
+                    diag.code === 'css-mediaqueryexpected' ||
+                    diag.code === 'css-ruleorselectorexpected'
+                ) {
+                    // caused by @st-scope with special characters (e.g. @st-scope [div=rtl])
+                    const offset = getOffsetFromPosition(meta.rawAst.toString(), diag.range.start);
+                    const [astNode, _astNodeOffset] = getAstNodeAt(meta.rawAst, offset + 1);
+                    const stScopeSanitized = 'media   ';
```

Comment:
We have to get rid of the native css lsp

Reply:
I knoooow.